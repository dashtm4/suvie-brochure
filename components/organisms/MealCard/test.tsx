import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'

import { MealCard } from '.'

describe('<MealCard />', () => {
  const meal = {
    id: 'bbq-salmon',
    title: 'BBQ Salmon',
    subtitle: 'with Southwest Brown Rice & Brussels Sprouts',
    imageSrc: 'bbq-salmon.jpg'
  }

  it('should renders correct <MealCard /> with minimum required meal fields', () => {
    const wrapper = shallow(<MealCard meal={meal} />)

    expect(wrapper.contains(meal.title)).toBeTruthy()
    expect(wrapper.contains(meal.subtitle)).toBeTruthy()

    expect(toJSON(wrapper)).toMatchSnapshot()
  })

  it('should renders correct <MealCard /> with all meal fields', () => {
    const meal = {
      id: 'bbq-salmon',
      title: 'BBQ Salmon',
      subtitle: 'with Southwest Brown Rice & Brussels Sprouts',
      description: 'Test description.',
      imageSrc: 'bbq-salmon.jpg',
      videoSrc: 'bbq-salmon.avi',
      posterSrc: 'bbq-salmon.jpg',
      type: 'your ingredients',
      prepTime: '10 minutes',
      calories: 500,
      categories: ['Carb Conscious', 'Gluten Free'],
      plateImageSrc: 'bbq-salmon_meal.png'
    }

    const wrapper = shallow(<MealCard meal={meal} />)

    expect(wrapper.contains(meal.title)).toBeTruthy()
    expect(wrapper.contains(meal.subtitle)).toBeTruthy()
    expect(wrapper.contains(meal.type)).toBeTruthy()
    expect(wrapper.contains(meal.categories[0])).toBeTruthy()
    expect(wrapper.contains(meal.categories[1])).toBeTruthy()

    expect(toJSON(wrapper)).toMatchSnapshot()
  })
})
