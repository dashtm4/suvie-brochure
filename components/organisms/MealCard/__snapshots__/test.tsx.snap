// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MealCard /> should renders correct <MealCard /> with all meal fields 1`] = `
<div
  className="relative w-60 max-w-full md:w-80"
  data-id="bbq-salmon"
>
  <figure
    className="relative max-w-full overflow-hidden rounded-lg shadow-md w-60 bg-secondary md:w-80 aspect-w-1 aspect-h-1"
    role="none"
  >
    <VideoPlayer
      autoplay={false}
      className="absolute inset-0 object-cover w-full h-full rounded-lg"
      controls={false}
      loop={true}
      muted={true}
      playerRef={
        Object {
          "current": null,
        }
      }
      playsInline={true}
      poster="bbq-salmon.jpg"
      url="bbq-salmon.avi"
    />
    <div
      className="absolute inset-0 w-full h-full rounded-lg"
      style={
        Object {
          "background": "linear-gradient(180deg, rgba(0, 0, 0, 0.5) 0%, rgba(0, 0, 0, 0) 100%)",
        }
      }
    />
    <button
      className="absolute inset-0 z-10 flex flex-col items-center justify-center w-full transition-all opacity-0 cursor-pointer md:hover:opacity-100"
      onClick={[Function]}
      role="presentation"
    >
      <Image
        alt=""
        height="28"
        src="/images/square-plus-icon.svg"
        width="28"
      />
      <div
        className="mt-2 font-bold tracking-widest text-white uppercase"
      >
        Learn more
      </div>
    </button>
    <div
      className="absolute left-0 bottom-auto flex flex-wrap justify-center w-full h-auto gap-1 px-4 top-4"
    >
      <div
        className="px-2 py-1 mx-1 text-xs font-bold tracking-widest text-center text-white uppercase border border-white rounded-lg md:py-2 md:px-4"
        key="Carb Conscious"
      >
        Carb Conscious
      </div>
      <div
        className="px-2 py-1 mx-1 text-xs font-bold tracking-widest text-center text-white uppercase border border-white rounded-lg md:py-2 md:px-4"
        key="Gluten Free"
      >
        Gluten Free
      </div>
    </div>
  </figure>
  <div
    className="flex flex-col items-center  text-center"
  >
    <div
      className="relative mb-2 mt-3"
      role="presentation"
    >
      <div
        className="py-2 px-4 font-bold tracking-widest text-center text-white uppercase whitespace-nowrap rounded-lg text-xs"
        style={
          Object {
            "backgroundColor": "#d0A86e",
          }
        }
      >
        your ingredients
      </div>
    </div>
    <h3
      className="mb-1 font-serif text-2xl font-normal text-secondary"
    >
      BBQ Salmon
    </h3>
    <div
      className="max-w-full text-gray-700 w-54"
    >
      with Southwest Brown Rice & Brussels Sprouts
    </div>
  </div>
</div>
`;

exports[`<MealCard /> should renders correct <MealCard /> with minimum required meal fields 1`] = `
<div
  className="relative w-60 max-w-full md:w-80"
  data-id="bbq-salmon"
>
  <figure
    className="relative max-w-full overflow-hidden rounded-lg shadow-md w-60 bg-secondary md:w-80 aspect-w-1 aspect-h-1"
    role="none"
  >
    <VideoPlayer
      autoplay={false}
      className="absolute inset-0 object-cover w-full h-full rounded-lg"
      controls={false}
      loop={true}
      muted={true}
      playerRef={
        Object {
          "current": null,
        }
      }
      playsInline={true}
      poster="bbq-salmon.jpg"
    />
    <div
      className="absolute inset-0 w-full h-full rounded-lg"
      style={
        Object {
          "background": "linear-gradient(180deg, rgba(0, 0, 0, 0.5) 0%, rgba(0, 0, 0, 0) 100%)",
        }
      }
    />
    <button
      className="absolute inset-0 z-10 flex flex-col items-center justify-center w-full transition-all opacity-0 cursor-pointer md:hover:opacity-100"
      onClick={[Function]}
      role="presentation"
    >
      <Image
        alt=""
        height="28"
        src="/images/square-plus-icon.svg"
        width="28"
      />
      <div
        className="mt-2 font-bold tracking-widest text-white uppercase"
      >
        Learn more
      </div>
    </button>
  </figure>
  <div
    className="flex flex-col items-center  text-center"
  >
    <div
      className="relative mb-2 mt-3"
      role="presentation"
    >
      <div
        className="py-2 px-4 font-bold tracking-widest text-center text-white uppercase whitespace-nowrap rounded-lg text-xs"
        style={
          Object {
            "backgroundColor": "#d0A86e",
          }
        }
      />
    </div>
    <h3
      className="mb-1 font-serif text-2xl font-normal text-secondary"
    >
      BBQ Salmon
    </h3>
    <div
      className="max-w-full text-gray-700 w-54"
    >
      with Southwest Brown Rice & Brussels Sprouts
    </div>
  </div>
</div>
`;
